pkg/instance/repository.go

```diff
@@ -21,6 +21,25 @@ type repository struct {
 	instanceParameterEncryptionKey string
 }
 
+func (r repository) DeleteDeployment(deployment *model.Deployment) error {
+	for _, instance := range deployment.Instances {
+		err := r.db.Unscoped().Delete(&model.DeploymentInstance{}, instance.ID).Error
```

Comment:
Can this also return a `gorm.ErrRecordNotFound` I see you are checking for it below